### DISCUSSION TITLE
Proposal for changes in DoniachModel

### DISCUSSION THREAD

**Discussion by JulioAzcarate** [lmfit/lmfit-py#709] [Ideas]
Hi!

I would like to suggest some changes to `DoniachModel`. 
The math expression actually used in `lmfit` is the same showed in CasaXPS, which is an approximation only valid for low-asymmetric peaks. That math expression lacks the factor "**Gamma(1-alpha)**" present in the original lineshape (eq. 18 in Doniach, S.; Sunjic, M. Many-Electron Singularity in X-Ray Photoemission and X-Ray Line Spectra from Metals. *Journal of Physics C: Solid State Physics* **1970**, 3 (2), 285–291. https://doi.org/10.1088/0022-3719/3/2/010.) Here, Gamma is the [Gamma function](https://en.wikipedia.org/wiki/Gamma_function)

Also, I would like to suggest the use of parameters names than the original paper:
| lmfit | Doniach-Sunjic paper|
|-------|------------------------------|
| `sigma`| `gamma`    (core-hole lifetime) |
|`gamma`| `alpha`    (singularity index, or asymmetry) |

Do you think these changes could be implemented in the next version of `lmfit`?

My best regards

**Comment by newville** [comment]
@JulioAzcarate that seems reasonable to me - I'm not an XPS expert, but happy to try to support that better.  Would you be willing to make a Pull Request? 

If our sigma is really meant to be "core hole lifetime", should that be a variable parameter or something that is looked up and used as a constant?  I guess another way to ask that is: are instrumental or other broadenings also Lorentzian?

**Reply by JulioAzcarate** [reply]
@newville Thank you so much for your answer and for considering it !!!. 

In XPS there is instrumental broadening. Then, the Lorentzian or Doniach-Sinjic (DS) function must be convolved with a gaussian function. Sometimes, the core-hole lifetime (Lorentzian width) and asymmetry are knowledge and must be set as constant during the fitting and just the Gaussian width is varied. In the case of Lorentzian-Gaussian convolution, one just uses a Voigt profile. However, in the case of DS profile, one must convolve the `DoniachModel` with a `GaussianModel` and the use of `sigma` parameter name in both functions could be confusing. For that reason, I propose changes in the names ... just to be more "*scholarship*".

If you considered improving the `DoniachModel` may be a good idea to use the convolved form DS-Gaussian (eq 7 and 8 in (1) Joyce, J. J.; Del Giudice, M.; Weaver, J. H. Quantitative Analysis of Synchrotron Radiation Photoemission Core Level Data. *Journal of Electron Spectroscopy and Related Phenomena* **1989**, 49 (1), 31–45. https://doi.org/10.1016/0368-2048(89)80035-0.).
For example, in IgorPro they implemented a function `dsgn` as the result of the DS-Gaussian convolution for the integration of eq 8 in Joyce's paper by Simpson's rule. Maybe a similar strategy could be done in `lmfit`. Then, the DS-Gaussian profile will have the following parameters:
| Parameters | meaning |
|--- | --- |
| `center` | center or position of the peak |
| `amplitude`| amplitude or height of the peak |
| `gamma`| Lorentzian width |
| `alpha` | asymmetry |
| `sigma` | Gaussian width |

Do you think it would be better to make a Pull Request proposing the `DoniachModel` as the DS-Gaussian convolution?

My best regards

**Comment by newville** [comment]
Hi Julio,

On Fri, Feb 12, 2021 at 7:02 AM JulioAzcarate <notifications@github.com>
wrote:

> @newville <https://github.com/newville> Thank you so much for your answer
> and for considering it !!!.
>
> In XPS there is instrumental broadening. Then, the Lorentzian or
> Doniach-Sinjic (DS) function must be convolved with a gaussian function.
> Sometimes, the core-hole lifetime (Lorentzian width) and asymmetry are
> knowledge and must be set as constant during the fitting and just the
> Gaussian width is varied. In the case of Lorentzian-Gaussian convolution,
> one just uses a Voigt profile.
>

Right, that would match my experience with other X-ray measurements.  I
don't think we want to get too domain-specific here, but have that
Lorentzian width be a constant makes sense while the Gaussian portion may
be more necessary to vary in a fit.

> However, in the case of DS profile, one must convolve the DoniachModel
> with a GaussianModel and the use of sigma parameter name in both
> functions could be confusing. For that reason, I propose changes in the
> names ... just to be more "*scholarship*".
>

When we picked the parameter names for the different peak-like models we
tried to use consistent names so that it would be easier to swap models.
For a lineshape/model class like DS that really does seem to be
domain-specific (or so widely in one domain that it dominates the others),
I think it would be OK to migrate the parameter names to match those most
commonly used in the literature.  That transition could cause some trouble,
but maybe a deprecation cycle or a new function model name would be
appropriate.

> If you considered improving the DoniachModel may be a good idea to use
> the convolved form DS-Gaussian (eq 7 and 8 in (1) Joyce, J. J.; Del
> Giudice, M.; Weaver, J. H. Quantitative Analysis of Synchrotron Radiation
> Photoemission Core Level Data. *Journal of Electron Spectroscopy and
> Related Phenomena* *1989*, 49 (1), 31–45.
> https://doi.org/10.1016/0368-2048(89)80035-0.).
> For example, in IgorPro they implemented a function dsgn as the result of
> the DS-Gaussian convolution for the integration of eq 8 in Joyce's paper by
> Simpson's rule. Maybe a similar strategy could be done in lmfit. Then,
> the DS-Gaussian profile will have the following parameters:
>

Sure, I think it would be fine to have a Model that convolved DS with a
Gaussian.

> Parameters means
> center center or position of the peak
> amplitude amplitude or height of the peak
> gamma Lorentzian width
> alpha asymmetry
> sigma Gaussian width
>
> Do you think it would be better to make a Pull Request proposing the
> DoniachModel as the DS-Gaussian convolution?
>

Yes, I think that is all OK.  That is, I guess that "convolved with a
Gaussian" is more general (well, "sigma -> Inf" would effectively be "not
convolved").

If others have suggestions or opinions on how to handle the transition,
please chime in.

--Matt
